tests/cxx/out_of_line.cc

```diff
@@ -0,0 +1,70 @@
+
+#include "out_of_line-dep.h"
+
+template <class T, class U = char>
+struct Class {
+  using type = T;
+
+  void method_in() { dependent(T{}); }
+  void method_out();
+
+  static void method_static_in() { dependent(T{}); }
+  static void method_static_out();
+};
+
+template <class T, class U>
+void Class<T, U>::method_out() {
+  dependent(T{});
+}
+
+template <class T, class U>
+void Class<T, U>::method_static_out() {
+  dependent(T{});
+}
+
+template <>
+void Class<int, int>::method_out() {
```

Comment:
As per my earlier comment, it would be interesting to see an explicit specialization with one of the specialized arguments as return type, something like:

```
// somewhere-else.h
struct Type {};

// template.h
Type Class<Type, int>::method_static_out() {
    return Type();
}
```
This won't work exactly as-is, because the inline implementation will require the complete type of `Type`, but it might be possible to do something tricky to provoke the case in question.

Reply:
I am not sure what I would be testing with that. I mean, the specialization is really like a regular method so the template logic wouldn't really trigger here.

Can you be more specific what we try to verify here?